ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+        key: "{{ ansible_ssh_user }}"
+    
+    - name: set homedir fact
+      set_fact:
+        homedir: "{{ getent_passwd | json_query('*[4]') | first }}"
+    
+    - name: Create directory to hold the registry files
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        mode: u=rwX,g=rX,o=rX
+        recurse: yes
+      loop:
+        - "{{ registry_dir_auth }}"
+        - "{{ registry_dir_cert }}"
+        - "{{ registry_dir_data }}"
+      become: true
+    
+    - name: Generate htpasswd entry
+      command: htpasswd -bBn dummy dummy
+      register: htpass_entry
+    
+    - name: Write htpasswd file
+      copy:
+        content: '{{ htpass_entry.stdout }}'
+        dest: "{{ registry_dir_auth }}/htpasswd"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: Create registry auth for pullsecret
+      set_fact:
+        disconnected_auth: '  {"{{ ansible_fqdn }}:{{ registry_port }}": {"auth": "ZHVtbXk6ZHVtbXk=", "email": "{{ ansible_user }}@redhat.com" } }'
+  
+    - name: Write auth for disconnected to registry host
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "{{ homedir }}/{{ registry_auth_file }}"
+  
+    - name: Write auth for disconnected to localhost
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "~/{{ registry_auth_file }}"
+      delegate_to: localhost
+  
+    - name: append auth to pullsecret
+      shell: |
+        echo '{{ pullsecret }}' | jq -c \
+          '.auths += {{ disconnected_auth }}'
+      register: new_pullsecret
+  
+    - name: set pullsecret with new auth
+      set_fact:
+        pullsecret: "  {{ new_pullsecret.stdout }}"
+  
+    - name: Write pullsecret to registry host
+      copy:
+        content: '{{pullsecret}}'
+        dest: "{{ homedir }}/pullsecret.txt"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: generate ssl certs
+      shell: |
+        openssl req \
+          -newkey rsa:4096 \
+            -nodes \
+            -sha256 \
+            -keyout {{ registry_dir_cert }}/domain.key \
+            -x509 -days 365 \
+            -out {{ registry_dir_cert }}/domain.crt \
+            -subj "/C={{ cert_country }}/ST={{ cert_state }}/L={{ cert_locality }}/O={{ cert_organization }}/OU={{ cert_organizational_unit }}/CN={{ ansible_fqdn }}"
+    
+    - name: Copy cert to pki directory
+      copy:
+        src: "{{ registry_dir_cert }}/domain.crt"
+        dest: /etc/pki/ca-trust/source/anchors/domain.crt
+        remote_src: yes
+        group: "{{ ansible_user }}"
+        owner: "{{ ansible_user }}"
+        mode: 0644
+        force: yes
+        backup: yes
+      become: true
+    
+    
+    - name: Update the CA trust files
+      command: update-ca-trust extract
+      become: true
+    
+    - name: Create container to serve the registry
```

Comment:
We could use the `podman_container.py` module here that we've added to the installer role's library instead of relying on command/shell.

Reply:
included. this will be pushed with my next commit

Reply:
I also had to remove the `become: true` from this task to get the container to work properly on my RHEL 8.1 baremetal prov host.  When running as a root container, I could not reach the registry.  I tried all sorts of iptables and firewalld magic in an effort to solve the problem, but nothing was efficacious.  Running as non-root made the registry endpoint reachable (without any manual firewall changes).

Reply:
i just removed it and my test just finished. I will push the change up with my next commit in a bit.

Reply:
pushed